x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/engines_list.tsx

```diff
@@ -95,7 +95,7 @@ export const EnginesList: React.FC = () => {
         isLoading={isLoading}
       >
         <EuiSpacer />
-        {results.length ? (
+        {searchQuery! || results?.length ? (
```

Comment:
This would remedy some problems that I created -sorry-, but I think we should update this page similar to how we did on the indices page. 
https://github.com/elastic/kibana/blob/main/x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_indices/indices_logic.ts#L177

Ideally we would move this loading to createApiLogic and get separate page loading from API loading statuses.

Reply:
Thanks Efe.  
Confused a bit though. We update the `isLoading` status based on the API loading status returned from createAPILogic in this [line](https://github.com/elastic/kibana/blob/main/x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/engines_list_logic.ts#L125)

Reply:
Sorry if I was not clear. 
What I meant is we should differentiate the page load and engines call which triggers each time when you type into the search box. We had similar problem and the solution at the indices page. 

In this code snippet below, you will see `isFirstRequest` which is a simple flag we use to differentiate in between page load and search calls. You can copy the implementation of that flag here to prevent both landing to the empty page and the flashing when you type.

https://github.com/elastic/kibana/blob/f3e96bf1e3f87ab1fe4111bdccf5ea4a8c2ff859/x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_indices/indices_logic.ts#L177-L179


The comment on moving this to the createApiLogic is an enhancements, that we can do it later.